buildroot/bin/build_all_examples

```diff
@@ -35,28 +44,31 @@ else
   echo "Resuming from $2"
 fi
 
-TMPDIR=`mktemp -d`
+# Create a temporary folder if needed
+[[ -d $MFTEMP ]] || export MFTEMP=$(mktemp -d)
 
 # Ok, let's do our stuff now
 # First extract the current temporary folder
 echo "Fetching configuration repository"
-if [ ! -e "$TMPDIR/README.md" ]; then
-  git clone --single-branch --branch "$1" https://github.com/MarlinFirmware/Configurations.git "$TMPDIR" || { echo "Failed to clone the configuration repository"; exit ; }
-  rm -r $TMPDIR/.git
+if [ ! -e "$MFTEMP/README.md" ]; then
+  git clone --depth 1 --single-branch --branch "$1" $GITREPO "$MFTEMP" || { echo "Failed to clone the configuration repository"; exit ; }
 fi
 
+echo "Temporary path used: $MFTEMP"
 echo
 echo "Start building now..."
 echo "====================="
 shopt -s nullglob
-for config in $TMPDIR/config/examples/*/; do
+for config in $MFTEMP/config/examples/*/; do
   [ -d "${config}" ] || continue
   base=`basename "$config"`
   if [ ! -z "$2" ] && [ "$2" != "$base" ]; then
     echo "Skipping $base..."
     continue
   fi
-  "$HERE/build_example" "internal" "$TMPDIR" "$base" || { echo "Failed to build $base"; exit ; }
+  "$HERE/build_example" "internal" "$MFTEMP" "$base" || { echo "Failed to build $base"; exit ; }
 done
 
-rm -r "$TMPDIR"
+# Delete the temp folder
+rm -rf "$MFTEMP"
```

Comment:
While I know what the script is doing, if, for whatever reason, MFTEMP get reassigned, this is a pending atomic bomb. Maybe adding a check here like this: `[[ -e $MFTEMP/.git ]] && rm -rf "$MFTEMP"` would be safer